src/main/java/seedu/saveit/logic/commands/EditCommand.java

```diff
@@ -139,23 +137,24 @@ private static Solution processNewSolution(int index, Issue issueToEdit, Solutio
         // if in the home directory, should not process this
         if (index >= issueToEdit.getSolutions().size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_ISSUE_DISPLAYED_INDEX);
-        } else {
-            Solution oldSolution = issueToEdit.getSolutions().get(index);
-            Solution updatedSolution;
+        }
 
-            String updatedSolutionLink =
-                newSolution.getLink().getValue().equals(DUMMY_SOLUTION_LINK) ? oldSolution.getLink().getValue()
-                    : newSolution.getLink().getValue();
-            String updatedSolutionRemark =
-                newSolution.getRemark().getValue().equals(DUMMY_SOLUTION_REMARK) ? oldSolution.getRemark().getValue()
-                    : newSolution.getRemark().getValue();
+        Solution oldSolution = issueToEdit.getSolutions().get(index);
+        Solution updatedSolution;
 
-            updatedSolution = new Solution(updatedSolutionLink, updatedSolutionRemark);
+        String updatedSolutionLink =
+            newSolution.getLink().getValue().equals(DUMMY_SOLUTION_LINK) ? oldSolution.getLink().getValue()
```

Comment:
I feel if there is a `EditSolutionDescriptor`, things will be much clearer.

Reply:
There is no need to store index in `EditIssueDescriptor` as well. (According to EditIssueDescriptor.java on my local computer)

Reply:
sure, will change in next PR. Hope I will not break code.

Reply:
thank you :)